lib/ansible/plugins/lookup/filetree.py

```diff
@@ -0,0 +1,123 @@
+# (c) 2016 Dag Wieers <dag@wieers.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+import pwd
+import grp
+import stat
+
+from ansible.plugins.lookup import LookupBase
+from __main__ import display
+warning = display.warning
+
+HAVE_SELINUX=False
+try:
+    import selinux
+    HAVE_SELINUX=True
+except ImportError:
+    pass
+
+def _to_filesystem_str(path):
+    '''Returns filesystem path as a str, if it wasn't already.
+
+    Used in selinux interactions because it cannot accept unicode
+    instances, and specifying complex args in a playbook leaves
+    you with unicode instances.  This method currently assumes
+    that your filesystem encoding is UTF-8.
+
+    '''
+    if isinstance(path, unicode):
+        path = path.encode("utf-8")
+    return path
+
+# If selinux fails to find a default, return an array of None
```

Comment:
This comes directly from the Ansible source code. Again, if this is general advice, let's fix it in Ansible first then get back to me with your PEP8 OCD.